### DISCUSSION TITLE
WAF CRS with keycloak

### DISCUSSION THREAD

**Discussion by syisunny** [keycloak/keycloak#35586] [Q&A]
Hello,
we are setting up a WAF in front of Keycloak. and it throws some false positive issues.
is there any recommendation to configure the paranoia level and the exclusion rule of CRS for keycloak?
Thank you very much in advance!

**Comment by itsmegrave** [comment]
I believe I have the same issue here.
Receiving some 403 from WAF when try to use `!@#` as part of a password.
Do you find any recommendation @syisunny

**Reply by syisunny** [reply]
unfortunately there is no perfect solution for it. we have to remove the rule one by one .
